src/DeviceInterface/AndroidInterface.js

```diff
@@ -40,9 +40,44 @@ class AndroidInterface extends InterfacePrototype {
         return Promise.resolve(userData)
     }
 
+    getEmailProtectionCapabilities () {
+        let deviceCapabilities = null
+
+        try {
+            deviceCapabilities = JSON.parse(window.EmailInterface.getDeviceCapabilities())
+        } catch (e) {
+            if (this.globalConfig.isDDGTestMode) {
+                console.error(e)
+            }
+        }
+
+        return Promise.resolve(deviceCapabilities)
+    }
+
     storeUserData ({addUserData: {token, userName, cohort}}) {
         return window.EmailInterface.storeCredentials(token, userName, cohort)
     }
+
+    removeUserData () {
+        try {
+            return window.EmailInterface.removeCredentials()
+        } catch (e) {
+            if (this.globalConfig.isDDGTestMode) {
+                console.error(e)
+            }
+        }
+    }
+
+    addLogoutListener (handler) {
+        // Cleanup on logout events
+        window.addEventListener('message', (e) => {
+            if (e.origin !== window.origin) return
+
+            if (e.data.emailProtectionSignedOut) {
+                handler()
+            }
+        })
+    }
```

Comment:
We need a way for the device to communicate to the web app when the user has been logged out -- open for discussion if this is the right approach, though it mirrors the listener / handler setup we have with the extension.

An alternative would be to have `removeUserData` return a promise so the page can refresh when complete, but this would mean we still wouldn't catch the user data is cleared from the device using native menus.

ℹ️  Also note: I've only checked this working on apple devices, not android.

Reply:
I'm not clear how these two work. Let's chat later. In any case, we should restrict the accepted origin to our domain (otherwise any website can send that message). This means that when initiated from the device menus we can't clear the autofill stuff right away (which is already the case). Maybe from the native layer they can trigger a page refresh or something?

Reply:
@alistairjcbrown as mentioned, these `message` handlers need to be restricted to the current domain only.

Reply:
@shakyShane They're restricted to the DuckDuckGo domain with the `this.globalConfig.isDDGDomain` -- do we want a `e.origin === window.origin` check in here too?